### PR TITLE
AMQP-678: Move BrokerRunning @Rule to junit Jar

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/AMQP-678

Convert `BrokerRunning` to use the amqp-client directly to avoid circular reference.

Also `LongRunningIntegrationTest`.

### CODE REVIEW

File: spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/BrokerRunning.java [hunk 1]
```diff
@@ -235,45 +227,146 @@ public Statement apply(Statement base, Description description) {
...
+	}
+
+	public void deleteExchanges(String... exchanges) {
+		ConnectionFactory connectionFactory = getConnectionFactory();
+		Connection connection = null;
+		Channel channel = null;
+
+		try {
+			connection = connectionFactory.newConnection();
+			connection.setId(generateId() + ".queueDelete");
```
Comment (artembilan) [inline spring-rabbit-junit/src/main/java/org/springframework/amqp/rabbit/junit/BrokerRunning.java]: `exchangeDelete` ?

File: src/reference/asciidoc/testing.adoc [hunk 2]
```diff
@@ -260,3 +260,43 @@ for the result.
...
+    brokerRunning.removeTestQueues("some.other.queue.too") // removes foo, bar as well
+}
+----
+
+Of course, there are times when you want tests to fail if there is no broker, such as a nightly CI build.
+To disable the rule at runtime, set an environment variable `RABBITMQ_SERVER_REQUIRED` to `true`.
+
+There are several `isRunning...` static methods such as `isBrokerAndManagementRunning()` which verifies the broker has the management plugin enabled.
+
+`LongRunningIntegrationTest` is a rule that disables long running tests; you might want to use this on a developer system but ensure that the rule is disabled on, for exaple, nightly CI builds.
```
Comment (artembilan) [inline src/reference/asciidoc/testing.adoc]: Typo: exa**m**ple

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/4fded3dde13a4d1e1f231a77ed4497e8d57b3321

**Comment by garyrussell** [comment]
Backport (1.7.x) PR: https://github.com/spring-projects/spring-amqp/pull/516

**Pull request closed without merging**